geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/list/AbstractLIndexIntegrationTest.java

```diff
@@ -0,0 +1,136 @@
+  }
+
+  @After
+  public void tearDown() {
+    flushAll();
+    jedis.close();
+  }
+
+  @Test
+  public void lindex_wrongNumberOfArgs_returnsError() {
+    assertExactNumberOfArgs(jedis, Protocol.Command.LINDEX, 2);
+  }
+
+  @Test
+  public void lindex_withPositiveIndex_withNonExistentList_returnsNull() {
+    assertThat(jedis.lindex(NON_EXISTENT_LIST_KEY, INDEX)).isNull();
+  }
+
+  @Test
+  public void lindex_withNegativeIndex_withNonExistentList_returnsNull() {
+    assertThat(jedis.lindex(NON_EXISTENT_LIST_KEY, -INDEX)).isNull();
+  }
+
+  @Test
+  public void lindex_withPositiveIndex_returnsElemnt() {
+    jedis.lpush(LIST_KEY, LIST_ELEMENTS);
+    assertThat(jedis.lindex(LIST_KEY, INDEX)).isEqualTo("deer");
+  }
+
+  @Test
+  public void lindex_withNegativeIndex_returnsElemnt() {
+    jedis.lpush(LIST_KEY, LIST_ELEMENTS);
+    assertThat(jedis.lindex(LIST_KEY, -INDEX)).isEqualTo("chameleon");
+  }
+
+  @Test
+  public void lindex_withPositiveOutOfRangeIndex_returnsNull() {
+    jedis.lpush(LIST_KEY, LIST_ELEMENTS);
+    assertThat(jedis.lindex(LIST_KEY, 10)).isNull();
+  }
+
+  @Test
+  public void lindex_withNegativeOutOfRangeIndex_returnsNull() {
+    jedis.lpush(LIST_KEY, LIST_ELEMENTS);
+    assertThat(jedis.lindex(LIST_KEY, -10)).isNull();
+  }
+
+  @Test
+  public void lindex_withInvalidIndex_withNonExistentList_returnsNull() {
+    assertThat(jedis.sendCommand(NON_EXISTENT_LIST_KEY, Protocol.Command.LINDEX,
+        NON_EXISTENT_LIST_KEY, "b")).isNull();
+  }
+
+  @Test
+  public void lindex_withInvalidIndex_returnsError() {
+    jedis.lpush(LIST_KEY, LIST_ELEMENTS);
+    assertThatThrownBy(() -> jedis.sendCommand(LIST_KEY, Protocol.Command.LINDEX, LIST_KEY, "b"))
+        .hasMessage("ERR " + ERROR_NOT_INTEGER);
+  }
+
+  @Test
+  public void lindex_withWrongKeyType_returnsWrongTypeError() {
+    String key = "{tag1}ding";
+    jedis.set(key, "dong");
+    assertThatThrownBy(() -> jedis.lindex(key, 2)).hasMessage("WRONGTYPE " + ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void lindex_withWrongKeyType_withInvalidIndex_returnsWrongTypeError() {
+    String key = "{tag1}ding";
+    jedis.set(key, "dong");
+    assertThatThrownBy(() -> jedis.sendCommand(LIST_KEY, Protocol.Command.LINDEX, key, "b"))
+        .hasMessage("WRONGTYPE " + ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void ensureListConsistency_whenRunningConcurrently() {
+    final AtomicReference<String> lindexResultReference = new AtomicReference<>();
+    new ConcurrentLoopingThreads(1000,
+        i -> jedis.lpush(LIST_KEY, LIST_ELEMENTS),
+        i -> lindexResultReference.set(jedis.lindex(LIST_KEY, -7)))
+            .runWithAction(() -> {
+              assertThat(lindexResultReference).satisfiesAnyOf(
+                  lindexResult -> assertThat(lindexResult.get()).isNull(),
+                  lindexResult -> assertThat(lindexResult.get()).isEqualTo("goat"));
+              jedis.del(LIST_KEY);
```

Comment:
This test might be a little bit better if instead of checking the concurrency behaviour between creating a new list via LPUSH and checking the index via LINDEX, it was checking the behaviour when adding a different set of elements to an already existing list, and checking that the last element is as expected. By checking the last element, we ensure that the index is never out of range, so we're more likely to catch any issues where we're reading the list while still adding elements to it.

Reply:
With the commands that are currently implemented, the only remove is lpop, which removes one element from the head. So I added only one element. I could change it so I add more then just a for loop to pop multiple added elements. 

When you add to the list it gets added to the head, so I checked the 0 index for the difference. If I checked the last element then for one of the cases it could be out of range.